lib/queue.js

```diff
@@ -46,255 +41,242 @@ FetchQueue.prototype.add = function(protocol, domain, port, path, depth, callbac
 
     var url = protocol + "://" + domain + (port !== 80 ? ":" + port : "") + path;
 
-    self.exists(protocol, domain, port, path,
-        function(err, exists) {
-            if (err) {
-                return callback(err);
-            }
-
-            if (!exists) {
-                var queueItem = {
-                    url: url,
-                    protocol: protocol,
-                    host: domain,
-                    port: port,
-                    path: path,
-                    depth: depth,
-                    fetched: false,
-                    status: "queued",
-                    stateData: {}
-                };
-
-                self.push(queueItem);
-                callback(null, queueItem);
-            } else {
-                var error = new Error("Resource already exists in queue!");
-                error.code = "DUP";
-
-                callback(error);
-            }
-        });
+    queue.exists(protocol, domain, port, path, function(err, exists) {
+        if (err) {
+            callback(err);
+        } else if (!exists) {
+            var queueItem = {
+                url: url,
+                protocol: protocol,
+                host: domain,
+                port: port,
+                path: path,
+                depth: depth,
+                fetched: false,
+                status: "queued",
+                stateData: {}
+            };
+
+            queue.push(queueItem);
+            callback(null, queueItem);
+        } else {
+            var error = new Error("Resource already exists in queue!");
+            error.code = "DUP";
+
+            callback(error);
+        }
+    });
 };
 
 // Check if an item already exists in the queue...
 FetchQueue.prototype.exists = function(protocol, domain, port, path, callback) {
-    callback = callback && callback instanceof Function ? callback : function() {};
-
+    callback = callback instanceof Function ? callback : noop;
     port = port !== 80 ? ":" + port : "";
 
     var url = (protocol + "://" + domain + port + path).toLowerCase();
 
     if (this.scanIndex[url]) {
         callback(null, 1);
-        return 1;
+    } else {
+        this.scanIndex[url] = true;
+        callback(null, 0);
     }
+};
 
-    this.scanIndex[url] = true;
-    callback(null, 0);
-    return 0;
-
+// Get item from queue
+FetchQueue.prototype.get = function(index, callback) {
+    callback = callback instanceof Function ? callback : noop;
+    callback(null, this[index]);
 };
 
 // Get last item in queue...
 FetchQueue.prototype.last = function(callback) {
```

Comment:
Originally it was intended to be part of the public API, in case anybody wanted to use it. If nobody is, we can get rid of it (but it's so hard to know. :-/)